erts/emulator/beam/jit/arm/instr_guard_bifs.cpp

```diff
@@ -1001,6 +987,88 @@ void BeamModuleAssembler::emit_bif_map_size(const ArgLabel &Fail,
     }
 }
 
+/* ================================================================
+ *  min/2
+ *  max/2
+ * ================================================================
+ */
+
+void BeamModuleAssembler::emit_bif_min_max(arm::CondCode cc,
+                                           const ArgSource &LHS,
+                                           const ArgSource &RHS,
+                                           const ArgRegister &Dst) {
+    auto [lhs, rhs] = load_sources(LHS, ARG1, RHS, ARG2);
+    auto dst = init_destination(Dst, ARG1);
+    bool both_small = always_small(LHS) && always_small(RHS);
+    bool need_generic = !both_small;
+    Label generic = a.newLabel(), next = a.newLabel();
+
+    if (both_small) {
+        comment("skipped test for small operands since they are always small");
+    } else if (always_small(RHS) &&
+               always_one_of(LHS, BEAM_TYPE_INTEGER | BEAM_TYPE_MASK_BOXED)) {
```

Comment:
`BEAM_TYPE_MASK_BOXED` includes all types that may be boxed, including some that may be immediates like pids and ports. You probably want `BEAM_TYPE_MASK_ALWAYS_BOXED` here.
```suggestion
               always_one_of(LHS, BEAM_TYPE_INTEGER | BEAM_TYPE_MASK_ALWAYS_BOXED)) {
```

Looking further into it, I see we have the same problem in `emit_is_lt` et al on `maint`. I'll make a PR that fixes that and uses `enum class` to prevent us from running into this problem again.

Reply:
Fixed.